src/AppInstallerRepositoryCore/Microsoft/PortableIndex.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PortableIndex.h"
+#include "SQLiteStorageBase.h"
+#include "Schema/MetadataTable.h"
+
+namespace AppInstaller::Repository::Microsoft
+{
+    PortableIndex PortableIndex::CreateNew(const std::string& filePath, Schema::Version version)
+    {
+        AICLI_LOG(Repo, Info, << "Creating new Portable Index [" << version << "] at '" << filePath << "'");
+        PortableIndex result{ filePath, version };
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(result.m_dbconn, "portableindex_createnew");
+
+        Schema::MetadataTable::Create(result.m_dbconn);

```

Comment:
This is code that shouldn't need to be repeated outside of the storage base.

Reply:
Moved to storage base constructor